frontend/src/screens/learner/details.tsx

```diff
@@ -0,0 +1,128 @@
+import { React, useState, useCallback, useNavigate, useParams } from '@common'
+import { ParticipantStudy, PublicResearcher } from '@api'
+import { LaunchStudy, isStudyLaunchable, StudyTopicID, StudyTopicTags, tagOfType } from '@models'
+import { useApi, dayjs } from '@lib'
+import { OffCanvas, Icon, IconKey, Box, Button } from '@components'
+import { colors } from '../../theme'
+
+interface StudyDetailsProps {
+    study: ParticipantStudy
+}
+
+const Part: React.FC<{ title: string, icon: IconKey }> = ({
+    children,
+    title,
+    icon,
+}) => {
+    if (!children) return null
+    return (
+        <Box direction="column">
+            <Box align='center' gap margin={{ vertical: 'default' }}>
+                <Icon icon={icon} color={colors.purple} />
+                <span css={{ color: colors.darkText }}>{title}</span>
+            </Box>
+            <div css={{ marginBottom: '0.5rem', color: colors.grayText }}>{children}</div>
+        </Box>
+    )
+}
+
+const StudyPart: React.FC<StudyDetailsProps & { title: string, icon: IconKey, property: string }> = ({
+    icon,
+    study,
+    title,
+    property,
+}) => {
+    const s = study as any
+    if (!s[property]) return null
+
+    return (
+        <Part title={title} icon={icon}>
+            {s[property]}
+        </Part>
+    )
+}
+
+const LaunchStudyButton: React.FC<StudyDetailsProps> = ({ study }) => {
+    const api = useApi()
+    const [isBusy, setBusy] = useState(false)
+
+    if (!isStudyLaunchable(study)) return null
+
+    const onLaunch = async () => {
```

Comment:
`useCallback`

Reply:
grr. I think that's the missing one that I deferred adding in the other PR. gonna do it now, thanks!